gomint-server/src/main/java/io/gomint/server/network/handler/PacketLoginHandler.java

```diff
@@ -147,171 +139,170 @@ public void handle(PacketLogin packet, long currentTimeMillis, PlayerConnection
+            // Default deny for maximum amount of players
+            if (server.onlinePlayers().size() >= server.serverConfig().maxPlayers()) {
+                event.cancelled(true);
+                event.kickMessage("Server is full");
+            }
 
-                this.server.pluginManager().callEvent(event);
-                if (event.cancelled()) {
-                    connection.disconnect(event.kickMessage());
-                    return;
-                }
+            server.pluginManager().callEvent(event);
+            if (event.cancelled()) {
+                connection.disconnect(event.kickMessage());
+                return;
+            }
 
-                if (this.keyFactory.keyPair() == null) {
-                    // No encryption
-                    connection.sendPlayState(PacketPlayState.PlayState.LOGIN_SUCCESS);
-                    connection.state(PlayerConnectionState.RESOURCE_PACK);
-                    connection.initWorldAndResourceSend();
-                } else {
-                    // Generating EDCH secrets can take up huge amount of time
-                    connection.server().executorService().execute(() -> {
-                        connection.server().watchdog().add(2, TimeUnit.SECONDS);
-
-                        // Enable encryption
-                        EncryptionHandler encryptionHandler = new EncryptionHandler(this.keyFactory);
-                        encryptionHandler.supplyClientKey(chainValidator.getClientPublicKey());
-                        if (encryptionHandler.beginClientsideEncryption()) {
-                            // Get the needed data for the encryption start
-                            connection.state(PlayerConnectionState.ENCRPYTION_INIT);
-
-                            byte[] key = encryptionHandler.key();
-                            byte[] iv = encryptionHandler.iv();
-
-                            // We need every packet to be encrypted from now on
-                            connection.inputProcessor().enableCrypto(key, iv);
-
-                            // Forge a JWT
-                            String encryptionRequestJWT = FORGER.forge(encryptionHandler.serverPublic(), encryptionHandler.serverPrivate(), encryptionHandler.clientSalt());
-
-                            // Tell the client to enable encryption after sending that packet we also enable it
-                            PacketEncryptionRequest packetEncryptionRequest = new PacketEncryptionRequest();
-                            packetEncryptionRequest.setJwt(encryptionRequestJWT);
-                            connection.send(packetEncryptionRequest, aVoid -> {
-                                connection.outputProcessor().enableCrypto(key, iv);
-                            });
-                        }
-
-                        connection.server().watchdog().done();
-                    });
+            // Update player world
+            player.world((WorldAdapter) player.spawnLocation().world());
```

Comment:
This should be synced up with the world

Reply:
waiting for https://github.com/gomint/gomint/pull/727#discussion_r571447875

Reply:
done